File: executor/src/main/java/org/example/entity/Result.java
package org.example.entity;

import lombok.Data;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
public class Result {

    @Id
    @GeneratedValue
    private Long id;
    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private List<ClassificationObjectStatistics> objectStatistics;
    private int k;
    private float splitRatio;
    private int testingSliceSize;
    private int trainingSliceSize;
    private float accuracy;
    private String algorithm;
    private LocalDateTime creationDate;
}

File: frontend/src/app/shared/result.ts
export interface Result {
  id?: number;
  objectStatistics: ClassificationObjectStatistics[];
  k: number;
  splitRatio: number;
  testingSliceSize: number;
  trainingSliceSize: number;
  accuracy: number;
  algorithm: string;
  creationDate: string;
}

export interface ClassificationObjectStatistics {
  id?: number;
  objectName: string;
  recall: number;
  fallout: number;
  precision: number;
}
